packages/block-library/src/columns/edit.js

```diff
@@ -78,8 +78,15 @@ function ColumnsEditContainer( {
 						value={ count }
 						onChange={ ( value ) => updateColumns( count, value ) }
 						min={ 2 }
-						max={ 6 }
+						max={ Math.max( 6, count ) }
 					/>
+					{ count > 6 && (
+						<Notice status="warning" isDismissible={ false }>
+							{ __(
+								'This number of columns used is too big. It may cause visual breakage.'
```

Comment:
The use of "big" here as it relates to the number is somewhat confusing, since it could be misinterpreted as "the columns themselves occupy more visual space".

Maybe something like:

```suggestion
								'This column count exceeds the recommended amount and may cause visual breakage.'
```